ports/nrf/modules/machine/temp.c

```diff
@@ -88,7 +88,11 @@ STATIC mp_obj_t machine_temp_read(mp_uint_t n_args, const mp_obj_t *args) {
     }
 #endif // BLUETOOTH_SD
 
-    return MP_OBJ_NEW_SMALL_INT(nrf_temp_read());
+    nrf_temp_task_trigger(NRF_TEMP, NRF_TEMP_TASK_START);
+    while (!nrf_temp_event_check(NRF_TEMP, NRF_TEMP_EVENT_DATARDY)) {
+    }
+
+    return MP_OBJ_NEW_SMALL_INT(nrf_temp_read() / 4);
```

Comment:
After we have read the value we should probably turn off the sampling to save some power.
`nrf_temp_task_trigger(NRF_TEMP, NRF_TEMP_TASK_STOP);
`

Reply:
[The documentation seems to imply that this isn't necessary](https://infocenter.nordicsemi.com/topic/ps_nrf52840/temp.html?cp=4_0_0_5_27). At least on the nRF52840

> When the temperature measurement is completed, TEMP analog electronics power down to save power.

> TEMP only supports one-shot operation, meaning that every TEMP measurement has to be explicitly started using the START task.

I'll implement it anyway though as it can't hurt to be sure